(dp1
S'output'
p2
S"<type 'exceptions.RuntimeError'> Table: missing required field: COORDINATES"
p3
sS'layer'
p4
S'/Users/hanli/SideProj/Server/web2py/applications/sf_parking/controllers/api.py'
p5
sS'code'
p6
S"@request.restful()\ndef rest_api():\n    def GET(table_name, id):\n    \tif not table_name == 'bike_parking':\n    \t\traise HTTP(404)\n        return dict(bike_parking = db.bike_parking(id))\n    def POST(*args,**vars):\n        if not args[0] == 'bike_parking':\n    \t\traise HTTP(404)\n    \treturn dict(bike_parking = db.bike_parking.validate_and_insert(**vars))\n    return locals()\nresponse._vars=response._caller(rest_api)\n"
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/Users/hanli/SideProj/Server/web2py/gluon/restricted.py", line 220, in restricted\n    exec ccode in environment\n  File "/Users/hanli/SideProj/Server/web2py/applications/sf_parking/controllers/api.py", line 12, in <module>\n  File "/Users/hanli/SideProj/Server/web2py/gluon/globals.py", line 385, in <lambda>\n    self._caller = lambda f: f()\n  File "/Users/hanli/SideProj/Server/web2py/gluon/globals.py", line 349, in f\n    return rest_action(*_self.args, **getattr(_self,\'vars\',{}))\n  File "/Users/hanli/SideProj/Server/web2py/applications/sf_parking/controllers/api.py", line 10, in POST\n    return dict(bike_parking = db.bike_parking.validate_and_insert(**vars))\n  File "/Users/hanli/SideProj/Server/web2py/gluon/dal.py", line 9131, in validate_and_insert\n    response.id = self.insert(**new_fields)\n  File "/Users/hanli/SideProj/Server/web2py/gluon/dal.py", line 9114, in insert\n    ret =  self._db._adapter.insert(self, self._listify(fields))\n  File "/Users/hanli/SideProj/Server/web2py/gluon/dal.py", line 9069, in _listify\n    \'Table: missing required field: %s\' % name)\nRuntimeError: Table: missing required field: COORDINATES\n'
p11
s.